Updates

barunio opened the pull request:
It would be particularly helpful to review the README and make sure it makes sense.

barunio committed:
update/ correct README

The Readme was using the wrong class name (SeedData instead of Seeder), 
which was a remnant from the initial transition. Also switched things over
to a more concise hash syntax.

barunio committed:
code rewrite, add specs

barunio committed:
improve README

barunio committed:
clean up gemspec and add Gemfile

barunio committed:
major version bump to 1.0

recent commits have resulted in an incompatible API change

jdewyea commented on README.md:
```diff
@@ -1,48 +1,91 @@
+  { data_type: "Water", name: "gallons", description: "Water usage" }
+]
+
+Seeder.create seeds, [:data_type, :name], DataField
```
Does `:description` need to be included in the array?

jdewyea replied:
Ah, nevermind. I was misreading the `create` method.

barunio replied:
Well, the fact that it was confusing means the README isn't as clear as it should be. One ought to be able to understand this without needing to look at the code. Any suggestions on how to modify this doc to make it clearer?

jdewyea replied:
I might add a line explaining that the second argument it takes is an array of one or more column names to use as the primary key for identifying existing records. That should be sufficient, I think.

jdewyea commented on lib/seeder.rb:
```diff
@@ -1,78 +1,66 @@
-      keys = @primary_keys[ar_model.name]
-      ar_model.all.map{|item| item.attributes.symbolize_keys.values_at(*keys) }
-    end
+  def initialize(data, keys, model)
```
Another minor suggestion -- I think from what I've seen in the rails source code the tendency is to use `model` to refer to an ActiveRecord instance, and use `model_class` to refer to the class itself. Not sure if it's worth changing that here, but I figured I'd mention it.

barunio replied:
Hm, I haven't seen any relevant code in the Rails source, but I'm going to keep this as is. I think the term `record` makes more sense for an ActiveRecord instance, and `model` is a good term for the model object, which is the class that a record is an instance of. `model_class` almost seems more like the parent class of the model object to me.

barunio commented:
@jdewyea I added a commit that clarifies the README a bit.

jdewyea commented:
:thumbsup:

barunio committed:
add TravisCI config file

fixlr commented:
:thumbsup:

barunio commented:
Huh, I'd completely forgotten about this PR :)

fixlr committed:
Add newest versions of Ruby and Rails to CI

fixlr committed:
Restrict mysql2 version to ~> 0.3.10

ActiveRecord expects mysql2 ~> 0.3.10 (and ~> 0.3.18) to be installed.
Now that mysql2 0.4.x has been released, we need to restrict the
versions of mysql2 that we test against.

This restriction can be relaxed once ActiveRecord is compatible

fixlr committed:
Drop unused tasks from Rakefile

fixlr committed:
Use new Travis features to speed up CI

* switch to the container infrastructure
* cache bundler
* use system libraries for nokogiri

barunio commented:
:rabbit: :rabbit2:
